### PR TITLE
refactor `fits` module

### PR DESCRIPTION
**Pull request by r4lv**

### CODE REVIEW

File: vip_hci/fits/fits.py [hunk 1]
```diff
@@ -190,20 +191,18 @@ def write_fits(filename, array, header=None, dtype32=True, verbose=True):
...
-        usually saved in float64 precision.
+    precision : numpy dtype, optional
+        Float precision, by default np.float32 or single precision float.
     verbose : bool, optional
         If True prints message.
 
     """
-    if dtype32:
-        array = array.astype('float32', copy=False)
+    array = array.astype(precision, copy=False)
```
Comment (carlgogo) [inline vip_hci/fits/fits.py]: Please also change/fix this in HCIDataset.save

Comment (r4lv) [reply vip_hci/fits/fits.py]: thanks, missed that!

File: vip_hci/fits/fits.py [hunk 2]
```diff
@@ -90,7 +91,7 @@ def open_adicube(fitsfilename, verbose=True):
     
     Parameters
     ----------
-    fitsfilename : string
+    fitsfilename : string or Path
         Name of the fits file.
```
Comment (carlgogo) [inline vip_hci/fits/fits.py]: I think it's worth explaining here the pathlib.Path option ;)

Comment (carlgogo) [reply vip_hci/fits/fits.py]: Hi @r4lv, I think you missed this comment. It's just changing the description of the fitsfilename. Same for open_fits

Comment (carlgogo) [reply vip_hci/fits/fits.py]: I mean to extend the description of the argument: Name of the fits file or .... object.

**Pull request merged**